apps/emqx_dashboard/test/emqx_swagger_SUITE.erl

```diff
@@ -0,0 +1,300 @@
+                <<"ip">> => {{127, 0, 0, 1}, 8081},
+                <<"note">> => "Update all gems",
+                <<"note_url">> => "string example", <<"oneof_test">> => 1,
+                <<"page">> => 1,
+                <<"scopes">> => ["public_repo", "user"],
+                <<"test">> => test},
+            headers =>
+            #{<<"accept">> => <<"application/json">>,
+                <<"accept-encoding">> => <<"gzip, deflate, br">>,
+                <<"accept-language">> =>
+                <<"zh-CN,zh;q=0.9,en;q=0.8,zh-TW;q=0.7">>,
+                <<"connection">> => <<"keep-alive">>,
+                <<"content-length">> => <<"181">>,
+                <<"content-type">> => <<"application/json">>,
+                <<"host">> => <<"127.0.0.1:18083">>,
+                <<"origin">> => <<"http://127.0.0.1:18083">>},
+            query_string =>
+            #{<<"is_admin">> => true,
+                <<"oneof_test_in_query">> => "a_good_oneof_in_query",
+                <<"per_page">> => 10}},
+    ?assertEqual({200}, do_parameters_validation(Request)),
+    ok.
+
+do_parameters_validation(Params) ->
+    Meta = #{module => ?MODULE, method => put,
+        path => "/test/clients/:client_id/:fingerprint"},
+    case emqx_dashboard_swagger:translate_req(Params, Meta) of
+        {error, Reason} ->
+            io:format("~p~n", [Reason]),
+            {401, 'BAD_REQUEST', Reason};
+        NewRequest ->
+            io:format("newRequest:~p~n", [NewRequest]),
+            {200}
+    end.
+
+paths() -> ["/test/clients/:client_id/:fingerprint"].
+
+schema("/test/clients/:client_id/:fingerprint") ->
+    #{
+        operationId => test,
+        put => #{
+            tags => [<<"oauth-authorizations">>],
+            description => ?DESC,
+            summary => <<"Get-or-create an authorization for a specific app and fingerprint">>,
+            parameters => [
+                {client_id, hoconsc:mk(string(), #{in => path, description => <<"The client ID of your Emqx app">>, example => <<"an long client id">>})},
+                {fingerprint, mk(emqx_schema:duration_s(), #{required => true, in => path, example => "12m"})},
+                {per_page, mk(range(1, 50), #{required => true, in => query, example => "10"})},
+                {is_admin, mk(boolean(), #{required => true, in => query, example => "true"})},
+                %% {good_ref, mk(hoconsc:ref(?MODULE, good_ref), #{required => true, in => query, example => "good ref out example"})}
+                {oneof_test_in_query, mk(hoconsc:union([string(), integer()]), #{in => query, example => "a_good_oneof_in_query"})}
+            ],
+            requestBody =>
+            #{
+                client_secret => mk(string(), #{description => <<"The OAuth app client secret for which to create the token.">>, maxLength => 40}),
+                scopes => mk(hoconsc:array(string()), #{<<"description">> => "A list of scopes that this authorization is in.", example => [
```

Comment:
question: when do we use `<<"description">>` and when do we  use `description` ?

Reply:
- input error. using `desc` is correct.
- yes we need to support both ways: `requestBody => ?REF(?MODULE, clients_fingerprint_request_body)` and no ref way.